Sources/Sentry/SentrySerialization.m

```diff
@@ -107,6 +107,34 @@ + (NSData *_Nullable)dataWithEnvelope:(SentryEnvelope *)envelope
     return envelopeData;
 }
 
++ (NSString *)baggageEncodedDictionary:(NSDictionary *)dictionary
+{
+    NSMutableArray *items = [[NSMutableArray alloc] initWithCapacity:dictionary.count];
+
+    NSMutableCharacterSet *allowedSet = [NSCharacterSet.alphanumericCharacterSet mutableCopy];
+    [allowedSet addCharactersInString:@"-_"];
+    NSInteger currentSize = 0;
+
+    for (id key in dictionary.allKeys) {
+        id value = dictionary[key];
+        NSString *keyDescription =
+            [[key description] stringByAddingPercentEncodingWithAllowedCharacters:allowedSet];
+        NSString *valueDescription =
+            [[value description] stringByAddingPercentEncodingWithAllowedCharacters:allowedSet];
+
+        NSString *item = [NSString stringWithFormat:@"%@=%@", keyDescription, valueDescription];
+        if (item.length + currentSize <= 8192) {
```

Comment:
`l`: We could break if we exceeded the max size

Reply:
That's why we have this check, to avoid exceeding the max size.

Reply:
I think @philipphofmann is suggesting we do not look at any more items once we reach max size. It could be possible though that there's smaller items that still fit even though the current loop item doesn't. So I think it's better this way.

Reply:
Ow, I see. @AbhiPrasad can correct me if Im wrong, but I remember reading somewhere from him that we should continue adding the values that fits. So, break is not the best solution.

Reply:
> If the resulting baggage-string would be greater than 8192 bytes, some list-members MAY be dropped until the resulting baggage-string is 8192 characters or less.

and

> If a platform does not propagate all list-members, it is left to the implementer to decide which list-members to propagate.

See https://www.w3.org/TR/baggage/#limits